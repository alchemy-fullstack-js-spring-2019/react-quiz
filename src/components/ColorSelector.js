import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class ColorSelector extends PureComponent {
  static propTypes = {
    addColors: PropTypes.func.isRequired
  };

  state = {
    color: '#000000',
    backgroundColor: '#FFFFFF'
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
    console.log(this.state);
    this.props.addColors(this.state);
  }

  render() {
    return (
      <>
        <p>
          text color
          <input type='color' name='color' value={this.state.color} onChange={this.handleChange} />
        </p>
        <p>
          background color 
          <input type='color' name='backgroundColor' value={this.state.backgroundColor} onChange={this.handleChange} />
        </p>
      </>
    );
  }

}
