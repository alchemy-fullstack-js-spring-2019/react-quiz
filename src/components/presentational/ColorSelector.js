import React from 'react';

export default function ColorSelector() {
  return (
    <input name="background-color" type="color" />,
    <input name="text-color" type="color" />
  );
}
