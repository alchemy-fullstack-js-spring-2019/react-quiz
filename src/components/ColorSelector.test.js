
import React from 'react';
import { shallow } from 'enzyme';
import ColorSelector from './ColorSelector';

describe('Color Input Component', () => {
  it('renders Color selector', () => {
    const wrapper = shallow(<ColorSelector selectColorHandler={jest.fn()} color={'blue'} backgroundColor={'green'}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
