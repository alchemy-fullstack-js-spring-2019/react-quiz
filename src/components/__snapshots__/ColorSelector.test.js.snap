// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelector ColorSelector 1`] = `
<Fragment>
  <p>
    text color
    <input
      name="color"
      onChange={[Function]}
      type="color"
      value="#000000"
    />
  </p>
  <p>
    background color
    <input
      name="backgroundColor"
      onChange={[Function]}
      type="color"
      value="#FFFFFF"
    />
  </p>
</Fragment>
`;
